Sources/MongoSwift/ConnectionString.swift

```diff
@@ -0,0 +1,82 @@
+import mongoc
+
+/// Class representing a connection string for connecting to MongoDB.
+internal class ConnectionString {
+    /// Pointer to the underlying `mongoc_uri_t`.
+    internal let _uri: OpaquePointer
+
+    /// Initializes a new `ConnectionString` with the provided options. Updates `options` to correctly reflect the
+    /// final options set on the `ConnectionString`.
+    internal init(_ connectionString: String, options: inout ClientOptions) throws {
```

Comment:
this internal api is pretty weird, but I assume the plan is to re-do it with the "rationalization" ticket. That's totally fine with me in order to get the client pooling stuff rolling.

Reply:
yeah, agreed. 
I thought about maybe adding a `ConnectionStringOptions` type and adding a property to `ConnectionString` to retrieve an instance of that that contained the most up to date options, or something like that. but I figured I would just leave that to figure out later.